external/avro/src/test/scala/org/apache/spark/sql/avro/AvroSuite.scala

```diff
@@ -896,4 +896,33 @@ class AvroSuite extends QueryTest with SharedSQLContext with SQLTestUtils {
       assert(count == 8)
     }
   }
+
+  test("SPARK-24881: write with compression - avro options") {
+    withTempPath { dir =>
+      val uncompressDir = s"$dir/uncompress"
+      val deflateDir = s"$dir/deflate"
+      val snappyDir = s"$dir/snappy"
+
+      val df = spark.read.format("avro").load(testAvro)
+      df.write
+        .option("compression", "uncompressed")
+        .format("avro")
+        .save(uncompressDir)
+      df.write
+        .options(Map("compression" -> "deflate", "compressionLevel" -> "9"))
+        .format("avro")
+        .save(deflateDir)
+      df.write
+        .option("compression", "snappy")
+        .format("avro")
+        .save(snappyDir)
+
+      val uncompressSize = FileUtils.sizeOfDirectory(new File(uncompressDir))
+      val deflateSize = FileUtils.sizeOfDirectory(new File(deflateDir))
```

Comment:
Thank you, @MaxGekk. Can we then check the type of compression at least `avro.codec	deflate`?